src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -65,22 +74,98 @@ public CCDCaseApi(
     }
 
     public Optional<Claim> getByExternalId(String externalId, String authorisation) {
-        final List<Claim> claims = search(authorisation, ImmutableMap.of("case.externalId", externalId));
+        return searchByExternalId(authorisation, externalId)
+            .map(CaseDetails::getData)
+            .map(this::convertToCCDCase)
+            .map((caseMapper::from));
+    }
 
-        if (claims.size() > 1) {
+    public void linkDefendant(String externalId, String defendantId, String authorisation) {
+        User user = userService.authenticateAnonymousCaseWorker();
+        CaseDetails caseResult = searchByExternalId(authorisation, externalId)
+            .orElseThrow(() -> new CoreCaseDataStoreException("Claim not found by external ID"));
+
+        caseAccessApi.grantAccessToCase(user.getAuthorisation(),
+            authTokenGenerator.generate(),
+            user.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            caseResult.getId() + "",
```

Comment:
What do you think about just calling `.toString()` on the ID explicitly?

Reply:
don't mind, this is CCD messing up their datatypes :(